scripts/ci/lib.sh

```diff
@@ -1352,6 +1357,22 @@ junit_wrap() {
     fi
 }
 
+junit_contains_failure() {
+    local dir="$1"
+    if [[ ! -d $dir ]]; then
+        return 1
+    fi
+    # There should be few files in such dir, and they should have well-behaved names,
+    # and "return" does not mix with piping to "while read", so we use a "for" over find.
+    # shellcheck disable=SC2044
```

Comment:
why we have this?

Reply:
Because of the unquoted variable expansion.